src/main/java/dev/joopie/jambot/api/youtube/ApiYouTubeService.java

```diff
@@ -9,75 +10,159 @@
+
+                return filteredVideos.stream().findFirst()
+                        .orElse(SearchResultDto.builder().found(false).build());
             } else {
-                log.warn("Invalid http response status ({}) returned.", response.getStatusLine().getStatusCode());
+                log.warn("Invalid HTTP response status: {}", response.getStatusLine().getStatusCode());
             }
-        } catch (IOException exception) {
-            log.warn("YouTube did an oepsie.", exception);
+        } catch (IOException e) {
+            log.warn("Error while fetching data from YouTube API", e);
         }
 
-        return SearchResultDto.builder()
-                .found(false)
-                .build();
+        return SearchResultDto.builder().found(false).build();
+    }
+
+    private List<String> getVideoIdsFromResponse(HttpEntity entity) throws IOException {
+        SearchResponse response = parseResponse(entity, SearchResponse.class);
+        return Optional.ofNullable(response)
+                .map(SearchResponse::getItems)
+                .orElse(List.of())
+                .stream()
+                .map(SearchResponse.Item::getId)
+                .map(SearchResponse.Item.Id::getVideoId)
+                .filter(videoId -> !trackSourceService.isRejected(videoId))
+                .toList();
+    }
+
+    private List<SearchResultDto> getFilteredVideos(List<String> videoIds, Duration minDuration, Duration maxDuration, List<String> artistNames) throws IOException {
+        String url = String.format(VIDEO_DETAILS_URL, String.join(",", videoIds), properties.getToken());
+        try (var client = HttpClients.createDefault();
+             var response = client.execute(RequestBuilder.get(url).build())) {
+
+            if (response.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+                List<SearchResponse.Item> items = getVideoDetailsFromResponse(response.getEntity());
+
+                List<SearchResultDto> matchingChannels = items.stream()
+                        .filter(item -> artistNames.stream().anyMatch(artist -> item.getSnippet().getChannelTitle().toLowerCase().contains(artist.toLowerCase())))
+                        .map(this::mapItemToSearchResult)
+                        .toList();
+
+                if (!matchingChannels.isEmpty()) {
+                    return matchingChannels;
+                }
+
+                return items.stream()
+                        .filter(item -> {
+                            Duration videoDuration = parseDuration(item.getContentDetails().getDuration());
+                            return videoDuration.compareTo(minDuration) >= 0 && videoDuration.compareTo(maxDuration) <= 0;
+                        })
+                        .sorted((item1, item2) -> {
+                            Duration duration1 = parseDuration(item1.getContentDetails().getDuration());
```

Comment:
Waarom geen getter methode in ContentDetails om een duration terug te geven?

Reply:
ContentDetails has a getter method?

Reply:
Die kun je maken ipv het elke keer het omt e zetten naar een Duration